src/RuntimeFeatureSettings.h

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright © 2015-2023 Synthstrom Audible Limited
+ *
+ * This file is part of The Synthstrom Audible Deluge Firmware.
+ *
+ * The Synthstrom Audible Deluge Firmware is free software: you can redistribute it and/or modify it under the
+ * terms of the GNU General Public License as published by the Free Software Foundation,
+ * either version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program.
+ * If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#ifndef RUNTIMEFEATURESETTINGS_H_
+#define RUNTIMEFEATURESETTINGS_H_
+
+#include <stdint.h>
+#include "MenuItemRuntimeFeatureSettings.h"
+#include "DString.h"
+#include "ResizeableArray.h"
+
+#define RUNTIME_FEATURE_SETTING_MAX_OPTIONS 8
+
+// State declarations
+enum RuntimeFeatureStateToggle : uint32_t { Off = 0, On = 1 };
+
+// Declare additional enums for specific multi state settings (e.g. like RuntimeFeatureStateTrackLaunchStyle)
+
+/// Every setting needs to be delcared in here
+enum RuntimeFeatureSettingType : uint32_t {
+	// FileFolderSorting // @TODO: Replace with actual identifier on first use
+	MaxElement // Keep as boundary
+};
+
+/// Definition for selectable options
+struct RuntimeFeatureSettingOption {
+	const char* displayName;
+	uint32_t value; // Value to be defined as typed Enum above
+};
+
+/// Every setting keeps its metadata and value in here
+struct RuntimeFeatureSetting {
+	const char* displayName;
+	const char* xmlName;
+	uint32_t value;
+	RuntimeFeatureSettingOption options[RUNTIME_FEATURE_SETTING_MAX_OPTIONS]; // Limited to safe memory
+};
+
+/// Encapsulating class
+class RuntimeFeatureSettings {
+public:
+	RuntimeFeatureSettings();
+
+	// Traded type safety for option values for code simplicity and size, use enum from above to compare
+	inline uint32_t get(RuntimeFeatureSettingType type) { return settings[type].value; };
+
+public:
+	void readSettingsFromFile();
+	void writeSettingsToFile();
+
+protected:
+	RuntimeFeatureSetting settings[RuntimeFeatureSettingType::MaxElement] = {
+
+	    //// @TODO: Remove example on first use
+	    // [RuntimeFeatureSettingType::FileFolderSorting] =  {
+	    //     .displayName = "File/Folder sorting",

```

Comment:
if this should be readable on 7 segment, names should be a single case alphanumeric only, name and the display should scroll like it does with file open if the filename is too long. Preferable the name should be very short. "FSORT" could be the better name in this example.

Reply:
The strings should be automatically displayed in the best readable casing and from the video I can confirm that they automatically scroll. Regarding length, for now this can be customized with a preporcessor ifdef in the definition, in the long run there was the idea to refactor displayable Strings into containers to also allow runtime switching. 

I can not Test with 7SEG but here is a link to the test video mentioned above https://discord.com/channels/608916579421257728/1107026299945496577/1120693871282757683. The menu had 3 test entries at the time